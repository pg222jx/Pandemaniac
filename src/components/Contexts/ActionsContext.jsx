import React,{ useState, createContext } from 'react';

export const ActionsContext = createContext()

export const ActionsProvider = props => {

    const [amount, setAmount] = useState(2)

    return (
        <ActionsContext.Provider value={[amount, setAmount]}>
            {props.children}
        </ActionsContext.Provider>
    )
}

