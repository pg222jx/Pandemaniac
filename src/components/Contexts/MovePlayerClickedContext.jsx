import React,{ useState, createContext } from 'react';

export const MovePlayerClickedContext = createContext()

export const MovePlayerClickedProvider = props => {

    const [clicked, setClicked] = useState(false)

    return (
        <MovePlayerClickedContext.Provider value={[clicked, setClicked]}>
            {props.children}
        </MovePlayerClickedContext.Provider>
    );
}

