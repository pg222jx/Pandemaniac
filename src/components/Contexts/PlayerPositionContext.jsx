import React,{ useState, createContext } from 'react';

export const PlayerPositionContext = createContext()

export const PlayerPositionProvider = props => {

    const [position, setPosition] = useState({bottom: undefined, left: undefined})

    return (
        <PlayerPositionContext.Provider value={[position, setPosition]}>
            {props.children}
        </PlayerPositionContext.Provider>
    )
}

