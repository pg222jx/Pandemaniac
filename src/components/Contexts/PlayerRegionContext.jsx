import React,{ useState, createContext } from 'react';

export const PlayerRegionContext = createContext()

export const PlayerRegionProvider = props => {

    const [region, setRegion] = useState('')

    return (
        <PlayerRegionContext.Provider value={[region, setRegion]}>
            {props.children}
        </PlayerRegionContext.Provider>
    )
}

