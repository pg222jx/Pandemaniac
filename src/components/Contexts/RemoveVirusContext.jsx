import React,{ useState, createContext } from 'react';

export const RemoveVirusContext = createContext()

export const RemoveVirusProvider = props => {

    const [clicked, setClicked] = useState(false)

    return (
        <RemoveVirusContext.Provider value={[clicked, setClicked]}>
            {props.children}
        </RemoveVirusContext.Provider>
    );
}

