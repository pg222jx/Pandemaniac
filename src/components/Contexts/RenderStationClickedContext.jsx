import React,{ useState, createContext } from 'react';

export const RenderStationClickedContext = createContext()

export const RenderStationClickedProvider = props => {

    const [clicked, setClicked] = useState(false)

    return (
        <RenderStationClickedContext.Provider value={[clicked, setClicked]}>
            {props.children}
        </RenderStationClickedContext.Provider>
    );
}

