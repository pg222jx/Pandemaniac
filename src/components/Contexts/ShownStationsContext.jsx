import React, { useState, createContext } from 'react'

export const ShownStationsContext = createContext()

export const ShownStationsProvider = props => {

    const [shownStations, setShownStations] = useState([])

    return (
        <ShownStationsContext.Provider value={[shownStations, setShownStations]}>
            {props.children}
        </ShownStationsContext.Provider>
    )
}

