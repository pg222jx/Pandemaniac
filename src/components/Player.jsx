/**
 * Representing the player.
 *
 * @author Pernilla GÃ¶th
 * @version 1.0.0
 */


import React, { useContext } from 'react'
import { PlayerPositionContext } from './Contexts/PlayerPositionContext'
import player from '../images/player.png'

const Player = () => {
    const [position, setPosition] = useContext(PlayerPositionContext)// eslint-disable-line no-unused-vars

    return (
        <img src={player} alt="The player" className="player" style={ position }/>
    )
}
 
export default Player
