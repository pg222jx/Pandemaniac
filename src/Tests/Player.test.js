import React from 'react'
import ReactDOM from 'react-dom'
import { render, screen } from '@testing-library/react';
import { shallow, mount } from 'enzyme'
import  { PlayerPositionContext, PlayerPositionProvider } from '../components/Contexts/PlayerPositionContext'
import Player from '../components/Player'

function renderPlayer(position, setPosition) {
  return render(
    <PlayerPositionContext.Provider value={[position, setPosition]}>
      <Player />
    </PlayerPositionContext.Provider>
  )
}

const wrapperShallow = shallow(
        <PlayerPositionProvider>
            <Player />
        </PlayerPositionProvider>
)

const wrapperMount = mount(
        <PlayerPositionProvider>
            <Player />
        </PlayerPositionProvider>
)

  describe('Testing the Player component', () => {
    it('should match the snapshot', () => {
        expect(wrapperShallow.html()).toMatchSnapshot()
      })

    it('should render without crashing', () => {
      const div = document.createElement('div')
      ReactDOM.render(
        <PlayerPositionProvider>
          <Player />
        </PlayerPositionProvider>, div
      )
     })

    test('so PlayerContext will set position', () => {
      const position = {bottom: '380px', left: '120px'}
      renderPlayer(position)
      expect(screen.getByAltText('The player').style.bottom).toEqual('380px')
      expect(screen.getByAltText('The player').style.left).toEqual('120px')
    })
        
    it('should have 1 img', () => {
      expect(wrapperMount.find('img').length).toEqual(1)
    })
  })