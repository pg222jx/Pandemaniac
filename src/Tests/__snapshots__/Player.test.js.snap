// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Player component should match the snapshot 1`] = `"<img src=\\"player.png\\" alt=\\"The player\\" class=\\"player\\"/>"`;
